src/components/App/App.js

```diff
@@ -1,26 +1,76 @@
 import React from 'react';
-import { PropTypes } from 'prop-types';
+import PropTypes from 'prop-types';
 import 'normalize.css/normalize.css';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import Snackbar from 'material-ui/Snackbar';
 
 import ComponentView from '../ComponentView/ComponentView';
 import Palette from '../Palette/Palette';
 import Footer from '../Footer/Footer';
 import VisibleHeader from '../../containers/VisibleHeader';
+import Modal from '../Modal/Modal';
+import getCodeConfigs from './codeConfigs';
 
 import './App.css';
 
+import getMuiTheme from '../../lib/getMuiTheme';
+import createCustomMuiTheme from '../../lib/createCustomMuiTheme';
+
 const propTypes = {
   palette: PropTypes.shape.isRequired,
+  isVisibleCodeModal: PropTypes.bool.isRequired,
+  showPaletteCodeModal: PropTypes.func.isRequired,
+  closePaletteCodeModal: PropTypes.func.isRequired,
+  isVisibleSnackBar: PropTypes.bool.isRequired,
+  snackBarText: PropTypes.string.isRequired,
+  showSnackBarWithText: PropTypes.func.isRequired,
+  closeSnackBar: PropTypes.func.isRequired,
 };
 
-const App = (props) => (
-  <div className="app">
-    <VisibleHeader />
-    <Palette palette={props.palette} />
-    <ComponentView palette={props.palette} />
-    <Footer />
-  </div>
-);
+const App = ({
+  palette,
+  showPaletteCodeModal,
+  closePaletteCodeModal,
+  isVisibleCodeModal,
+  isVisibleSnackBar,
+  snackBarText,
+  showSnackBarWithText,
+  closeSnackBar,
+}) => {
+  const codeConfigs = getCodeConfigs(palette);
+
+  function closeModal(isSavedCode) {
+    closePaletteCodeModal();
+    if (isSavedCode) {
+      showSnackBarWithText('Code saved in buffer');
+    }
+  }
+
+  const modal = isVisibleCodeModal
+    ? <Modal content={codeConfigs} open={isVisibleCodeModal} onClose={(isSavedCode) => closeModal(isSavedCode)} />
+    : '';
+  const theme = createCustomMuiTheme(palette);
+
+  return (
+    <MuiThemeProvider muiTheme={getMuiTheme(theme)}>
```

Comment:
Let's use a standalone theme provider for non-palette components.
e.g. remove `muiTheme={...}` here and keep it in ComponentView.